src/Components/Aspire.Elastic.Clients.Elasticsearch/ElasticsearchHealthCheckBuilderExtensions.cs

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Elastic.Clients.Elasticsearch;
+/// <summary>
+/// Extension methods to configure <see cref="ElasticsearchHealthCheck"/>.
+/// </summary>
+internal static class ElasticsearchHealthCheckBuilderExtensions
+{
+    private const string NAME = "elasticsearch";
+
+    /// <summary>
+    /// Add a health check for Elasticsearch databases.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="elasticsearchUri">The Elasticsearch connection string to be used.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'elasticsearch' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddElasticsearch(
```

Comment:
Can we trim the unused code? Hopefully the new HealthCheck package is released soon and we can just use that.

@adamsitnik @unaizorrilla @sungam3r - any thoughts on when a new https://www.nuget.org/packages/AspNetCore.HealthChecks.Elasticsearch version can be released with https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2244?

Reply:
Hmm since it's a breaking change, we would need to release first `9.0-*` package (rather than just 8.0.2). To do that we should most likely update all MS and System dependencies to `9.0-`. It's not hard, but time consuming (it should take less time than last year with https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2044 since I've refactored the yaml files).

@eerhardt how important is this to you? If it's not very important, I would prefer to wait for RC1 bits being available at nuget.org.

cc @unaizorrilla

Reply:
> To do that we should most likely update all MS and System dependencies to 9.0-.

We should have a conversation about that. Since .NET 9 isn't LTS, we don't want to force Aspire users to move to net9.0 TFM just to take advantage of new Aspire features. The current plan (from my understanding) is that Aspire 9.0 will target net8.0 TFM, so users can use either net8.0 or net9.0 in their projects. This is similar to how efcore operates - it targets the last LTS.

* https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/9.0.0-preview.6.24327.4#dependencies-body-tab
* https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/8.0.7#dependencies-body-tab
* https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/7.0.20#dependencies-body-tab
* https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/6.0.32#dependencies-body-tab

Notice that `9.0.0-preview.6.24327.4` and `8.0.7` target `net8.0` while `7.0.20` and `6.0.32` target `net6.0`.